jdi-light/src/main/java/com/epam/jdi/light/asserts/generic/table/IDataGridAssert.java

```diff
@@ -0,0 +1,266 @@
+    }
+    /**
+     * Check that the table has row that has expected data
+     * @param data to compare
+     * @return DataTableAssert
+     */
+    @JDIAction("Assert that '{name}' has {0}")
+    public IDataGridAssert<D, T, A> row(D data) {
+        return row(d -> d.equals(data));
+    }
+    @JDIAction("Assert that '{name}' has {0}")
+    public IDataGridAssert<D, T, A> value(D expected, Row row) {
+        return value(d-> d.equals(expected), row);
+    }
+    @JDIAction("Assert that '{name}' rows {0}")
+    public IDataGridAssert<D, T, A> rows(Matcher<? super List<D>> condition) {
+        jdiAssert(dataGrid().allData(), condition);
+        return this;
+    }
+    @JDIAction("Assert that '{name}' is sorted")
+    public IDataGridAssert<D, T, A> dataSortedBy(JFunc2<D, D,Boolean> condition) {
+        List<D> allRows = dataGrid().allData();
+        for (int i = 1; i < allRows.size(); i++)
+            if (!condition.execute(allRows.get(i-1), allRows.get(i)))
+                jdiAssert("Table is not sorted at "+i+" row", Matchers.is(""));
+        jdiAssert("Table is sorted", Matchers.is("Table is sorted"));
+        return this;
+    }
+
+    public Compare exact(int count) {
+        return new Compare(count, this, EXACT);
+    }
+    public Compare atLeast(int count) {
+        return new Compare(count, this, ATLEAST);
+    }
+    public Compare no() {
+        return exact(0);
+    }
+    public Compare all() {
+        return new Compare(this);
+    }
+    public Compare onlyOne() {
+        return exact(1);
+    }
+
+    public enum CompareType {
+        EXACT("exactly"), ATLEAST("at least"), ALL("all");
+        public String text;
+        CompareType(String text) { this.text = text; }
+    }
+
+    public class Compare implements JAssert {
+
+        public int count;
+        public String name;
+        public String printText;
+        IDataGridAssert<D, T, A> dtAssert;
+        CompareType compareType;
+        public JDIBase base() { return IDataGridAssert.this.base(); }
+
+        private Compare(IDataGridAssert<D, T, A> dtAssert) {
+            this.dtAssert = dtAssert;
+            this.compareType = ALL;
+            this.printText = "all rows";
+            this.name = dtAssert.name;
+        }
+        private Compare(int count, IDataGridAssert<D, T, A> dtAssert, CompareType compareType) {
+            this.count = count;
+            this.dtAssert = dtAssert;
+            this.compareType = compareType;
+            this.printText = format("has %s %s rows that", compareType.text, count);
+            this.name = dtAssert.name;
+        }
+
+        /**
+         * Check that the table has rows that meet expected condition
+         * @param condition to compare
+         * @return DataTableAssert
+         */
+        @JDIAction("Assert that '{name}' {printText} meet expected condition")
+        public IDataGridAssert<D, T, A> rows(JFunc1<D, Boolean> condition) {
+            switch (compareType) {
```

Comment:
Codacy found an issue: [Switch statements should have a default label](https://app.codacy.com/gh/jdi-testing/jdi-light/pullRequest?prid=6397461)